openedx/core/djangoapps/external_user_ids/signals.py

```diff
@@ -0,0 +1,40 @@
+"""
+Signal Handlers for External User Ids to be created and maintainer
+"""
+
+from logging import getLogger
+
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+from student.models import CourseEnrollment
+
+from openedx.core.djangoapps.catalog.utils import get_programs
+from .models import ExternalId, ExternalIdType
+
+LOGGER = getLogger(__name__)
+
+
+@receiver(post_save, sender=CourseEnrollment)
```

Comment:
Thought: I don't see a use of it yet in platform, but you can set `sender` to the full path of the model. `"student.models.CourseEnrollment"` if you wanted to skip a direct import from another classes models. I don't think it particularly matters here..., but just a thought.